apps/menu-bar/src/components/TrustedDomainsInput.tsx

```diff
@@ -0,0 +1,80 @@
+import { darkTheme, lightTheme } from '@expo/styleguide-native';
+import React, { useState } from 'react';
+import { StyleSheet, TextInput as NativeTextInput } from 'react-native';
+
+import { TextInput } from './Text';
+import { Row } from './View';
+import { PlatformColor } from '../modules/PlatformColor';
+import { addOpacity } from '../utils/theme';
+import { useCurrentTheme } from '../utils/useExpoTheme';
+
+const TrustedDomainsInput = React.forwardRef<
+  NativeTextInput,
+  React.ComponentProps<typeof TextInput> & {
+    initialValue?: string[];
+    onSave: (value: string[]) => void;
+  }
+>(({ editable, initialValue, onSave, ...props }, forwardedRef) => {
+  const theme = useCurrentTheme();
+  const [value, setValue] = useState(initialValue?.join(', ') ?? '');
+
+  const backgroundColor =
+    theme === 'light'
+      ? addOpacity(lightTheme.background.default, 0.6)
+      : addOpacity(darkTheme.background.default, 0.2);
+
+  const handleSave = () => {
+    const formattedValue = value
+      ?.split(',')
+      .map((domain) => domain.trim())
+      .filter((i) => !!i);
+
+    onSave(formattedValue ?? []);
+  };
+
+  return (
+    <Row
+      border="light"
+      rounded="medium"
+      align="center"
+      style={[styles.inputContainer, { backgroundColor }, !editable && styles.inputDisabled]}>
+      <TextInput
+        shadow="input"
+        {...props}
+        value={value}
+        style={styles.input}
+        placeholderTextColor={PlatformColor('placeholderTextColor')}
+        ref={forwardedRef as any}
+        editable={editable}
+        onChangeText={setValue}
+        onSubmitEditing={handleSave}
```

Comment:
this is triggered when we press `enter` but can put a save button or something like that to make it more obvious.